storage/localstore/mode_set.go

```diff
@@ -214,6 +214,38 @@ func (db *DB) setSync(batch *leveldb.Batch, addr chunk.Address) (gcSizeChange in
 	}
 	item.AccessTimestamp = now()
 	db.retrievalAccessIndex.PutInBatch(batch, item)
+
+	// moveToGc toggles the deletion of the item from pushsync index
+	// it will be false in the case pull sync was called but push sync was meant on that chunk (!tag.Anonymous)
+	moveToGc := false
+
+	if db.tags != nil {
+		i, err = db.pushIndex.Get(item)
+		switch err {
+		case nil:
+			tag, _ := db.tags.Get(i.Tag)
+			if tag != nil {
+				switch mode {
+				case chunk.ModeSetSyncPull:
+					if tag.Anonymous {
+						// this will not get called twice because we remove the item in L217
```

Comment:
Line number seems not correct. Maybe to use different relations than line numbers?